crates/sncast/src/state/state_file.rs

```diff
@@ -40,17 +109,71 @@ impl ScriptTransactionEntries {
     pub fn get(&self, key: &str) -> Option<&ScriptTransactionEntry> {
         self.transactions.get(key)
     }
+
+    pub fn insert(&mut self, key: &str, entry: ScriptTransactionEntry) {
+        self.transactions.insert(key.to_string(), entry);
+    }
+
+    #[must_use]
+    pub fn get_success_output(&self, tx_id: &str) -> Option<ScriptTransactionOutput> {
+        if let Some(entry) = self.get(tx_id) {
+            if entry.status == ScriptTransactionStatus::Success {
+                return Some(entry.output.clone());
+            }
+        }
+        None
+    }
 }
 
 #[derive(Clone, Deserialize, Serialize, Debug, PartialEq)]
 pub struct ScriptTransactionEntry {
     pub name: String,
```

Comment:
Why this is `name` instead of `type`?

Reply:
`type` is a reserved keyword. besides, it marks a function `name`, not its type (in a contexts of script)

Reply:
In the state file, it shows a structure like `transaction` -> `name` -> for instance `declare`. Using `name` makes sense within the context of functions but is less fitting for transactions. Consider this as a nitpick.

Reply:
Hmm, I see your point, as this is a nitpick let's consider this under https://github.com/foundry-rs/starknet-foundry/issues/1968 as I'm not sure on the impact it may have